### DISCUSSION TITLE
[Potential unintended behavior] Github Pages won't render /README.html page for README.md

### DISCUSSION THREAD

**Discussion by yao-yin** [community/community#30162] [Other Feature Feedback, Questions, & Ideas]
Hi,

As the title says, Github Pages won't render a `/README.html` page for `README.md` file. This is usually fine because Github Pages will render it as the entry point, i.e., if we directly visit `<domain>/<some path>`, we could visit the page rendered from `<some path>/README.md`.

The subtle thing is from [relative link](https://github.blog/2013-01-31-relative-links-in-markup-files/). If we want to create a relative link to a README.md file, the syntax might look like:
```
[a relative link](<some path>/README.md)
```
And when we derive the relative link on the GitHub Pages, the link will be pointed to the url: `<domain>/<some path>/README.html` according to the [jekyll relative links](https://github.com/benbalter/jekyll-relative-links). That would cause a broken link.

A potential fix up is to remove the `README.md` part from the relative link, like:
```
[a relative link](<some path>)
```

But in that case, we can't jump to the README.md file in the GitHub repository.

A concrete example might help to understand the dilemma:

This broken link (https://jumpcrypto.github.io/crypto-reading-list/L1/README.html) is derived from [here](https://github.com/JumpCrypto/crypto-reading-list/blob/main/README.md?plain=1#L163).
```
* [Historical/academic background](L1/README.md)
```
If we change it to 
```
* [Historical/academic background](L1)
```
The broken link issue will be mitigated, but we can't jump to the README.md file in the repository. [Example](https://github.com/yao-yin/crypto-reading-list/blob/main/README.md?plain=1#L163).

The potential fix could be simple if we also render a README.html page for the README.md file.

Have a nice day,
Yin

**Comment by yoannchaudet** [comment]
Sounds like maybe you want to open an issue in https://github.com/benbalter/jekyll-readme-index? I have a feeling it's a mix of these two plugins interacting between each other and causing you troubles in Jekyll.
